api.go

```diff
@@ -19,37 +19,41 @@ package rocketmq
 
 import (
 	"context"
-
+	"github.com/apache/rocketmq-client-go/consumer"
 	"github.com/apache/rocketmq-client-go/primitive"
+	"github.com/apache/rocketmq-client-go/producer"
 )
 
 type Producer interface {
 	Start() error
 	Shutdown() error
-	SendSync(context.Context, ...*primitive.Message) (primitive.SendResult, error)
-	SendAsync(context.Context, func(primitive.SendResult), ...*primitive.Message) error
-	SendOneWay(context.Context, ...*primitive.Message) error
+	SendSync(context.Context, *primitive.Message) (*primitive.SendResult, error)
+	SendAsync(context.Context, func(context.Context, *primitive.SendResult), *primitive.Message) error
```

Comment:
func(context.Context, *primitive.SendResult) -> func(context.Context, *primitive.SendResult, error) is better because of the error of the receiving message.

Reply:
done